components/eam/src/physics/cam/ndrop.F90

```diff
@@ -1084,20 +868,349 @@ subroutine dropmixnuc( &
+         fluxm,flux_fullact)
+
+
+end subroutine get_activate_frac
+
+!===============================================================================
+
+subroutine update_from_newcld(cldn_in,cldo_in,dtinv,   &
+                wtke_in,temp_in,cs_kk_in,state_q_in,  &
+                qcld_in,raercol_pt,raercol_cw_pt, &
+                nsource_out, factnum_out)
+
+   use modal_aero_data,   only:   lmassptr_amode, numptr_amode
+
+   ! input arguments
+   real(r8), intent(in) :: cldn_in   ! cloud fraction [fraction]
+   real(r8), intent(in) :: cldo_in   ! cloud fraction on previous time step [fraction]
+   real(r8), intent(in) :: dtinv     ! inverse time step for microphysics [s^{-1}]
+   real(r8), intent(in) :: wtke_in   ! subgrid vertical velocity [m/s]
+   real(r8), intent(in) :: temp_in   ! temperature [K]
+   real(r8), intent(in) :: cs_kk_in  ! air density at actual level kk [kg/m^3]
+   real(r8), intent(in) :: state_q_in(:) ! aerosol mmrs [kg/kg]
+
+   real(r8), intent(inout) :: qcld_in
+   real(r8), intent(inout) :: nsource_out
+   real(r8), intent(inout) :: raercol_pt(:)
+   real(r8), intent(inout) :: raercol_cw_pt(:)
+   real(r8), intent(inout) :: factnum_out(:)
+
+!  local variables
+
+   real(r8) :: delt_cld        ! new - old cloud fraction [fraction]
+   real(r8) :: frac_delt_cld        ! fractional change in cloud fraction [fraction]
+   integer  :: imode       ! mode counter variable
+   integer  :: lspec       ! species counter variable
+   integer  :: mm                  ! local array index for MAM number, species
+   real(r8) :: dact             ! cloud-borne aerosol tendency due to cloud frac tendency [#/kg or kg/kg]
+   real(r8) :: fn(ntot_amode)              ! activation fraction for aerosol number [fraction]
+   real(r8) :: fm(ntot_amode)              ! activation fraction for aerosol mass [fraction]
+   integer  :: num_idx  ! number index
+   integer  :: spc_idx  ! species index
+   real(r8) :: flux_fullact ! flux of activated aerosol fraction assuming 100% activation [m/s]
+   real(r8) :: fluxn(ntot_amode)     ! flux of activated aerosol number fraction into cloud [m/s]
+   real(r8) :: fluxm(ntot_amode)     ! flux of activated aerosol mass fraction into cloud [m/s]
+
+
+         ! shrinking cloud ......................................................
+         !    treat the reduction of cloud fraction from when cldn(i,k) < cldo(i,k)
+         !    and also dissipate the portion of the cloud that will be regenerated
+
+         if (cldn_in < cldo_in) then
+            !  droplet loss in decaying cloud
+            !++ sungsup
+            nsource_out = nsource_out + qcld_in*(cldn_in - cldo_in)/cldo_in*dtinv
+            qcld_in      = qcld_in*(1._r8 + (cldn_in - cldo_in)/cldo_in)
+            !-- sungsup
+
+            ! convert activated aerosol to interstitial in decaying cloud
+
+            delt_cld = (cldn_in - cldo_in)
+            frac_delt_cld = delt_cld / cldo_in
```

Comment:
Will try to do this.  Last time though, I tried to do something like this and got a non-BFB error.  But maybe if I try more carefully it will work.  Agree this makes more sense.